lib/commands/device/index.ts

```diff
@@ -76,33 +81,45 @@ export default class DeviceCmd extends Command {
 
 		const balena = getBalenaSdk();
 
-		const device = (await balena.models.device.get(params.uuid, {
-			$select: [
-				'device_name',
-				'id',
-				'overall_status',
-				'is_online',
-				'ip_address',
-				'mac_address',
-				'last_connectivity_event',
-				'uuid',
-				'supervisor_version',
-				'is_web_accessible',
-				'note',
-				'os_version',
-				'memory_usage',
-				'memory_total',
-				'public_address',
-				'storage_block_device',
-				'storage_usage',
-				'storage_total',
-				'cpu_usage',
-				'cpu_temp',
-				'cpu_id',
-				'is_undervolted',
-			],
-			...expandForAppName,
-		})) as ExtendedDevice;
+		const device = (await balena.models.device.get(
+			params.uuid,
+			options.json
+				? {
+						$expand: {
+							device_tag: {
+								$select: ['tag_key', 'value'],
+							},
+							...expandForAppName.$expand,
+						},
+				  }
+				: {
+						$select: [
+							'device_name',
+							'id',
+							'overall_status',
+							'is_online',
+							'ip_address',
+							'mac_address',
+							'last_connectivity_event',
+							'uuid',
+							'supervisor_version',
+							'is_web_accessible',
+							'note',
+							'os_version',
+							'memory_usage',
+							'memory_total',
+							'public_address',
+							'storage_block_device',
+							'storage_usage',
+							'storage_total',
+							'cpu_usage',
+							'cpu_temp',
+							'cpu_id',
+							'is_undervolted',
+						],
+						...expandForAppName,
+				  },
```

Comment:
Tiny suggestion so that the `$select` is still used when the `--json` flag is used.
```suggestion
			{
				$select: [
					'device_name',
					'id',
					'overall_status',
					'is_online',
					'ip_address',
					'mac_address',
					'last_connectivity_event',
					'uuid',
					'supervisor_version',
					'is_web_accessible',
					'note',
					'os_version',
					'memory_usage',
					'memory_total',
					'public_address',
					'storage_block_device',
					'storage_usage',
					'storage_total',
					'cpu_usage',
					'cpu_temp',
					'cpu_id',
					'is_undervolted',
				],
				$expand: {
					...expandForAppName.$expand,
					...(options.json && {
						device_tag: {
							$select: ['tag_key', 'value'],
						},
					})
				},
			},
```

Reply:
Alternatively in case not including the $select was intentional, we could make clear that it was intentional by wrapping it in a `...(!options.json && { $select: [/*...*/] }),` along w/ a comment.

Reply:
@thgreasi thanks for the review. 

It was intentional not to limit the json response based on the fields used by the table printer. JSON format in other tools is a raw data dump that allows the consumer to determine what data is important, which is one of the primary use cases for it. (See also similar discussion in my first PR). 

I wouldn't personally add anything to the code because I think the intent of the code is clear based on the use case and experience, but I respect that it's y'all's codebase and I think you can commit that comment to it if you want. 

Hope that helps.